### PR TITLE
GITC-640: error handling in cart

### PR DESCRIPTION
**Pull request by nutrina**
##### Description

Note: this PR depends on #9808

GITC-640 Have improved error handling, according to design.

##### Refers/Fixes

GITC-640

##### Testing

https://user-images.githubusercontent.com/8137830/145029028-cbd3d911-8ecb-4611-b88d-06d708421edd.mov

https://user-images.githubusercontent.com/8137830/145029036-3db26de3-491f-4dde-825d-da8daa3d2e14.mov

### CODE REVIEW

File: app/grants/templates/grants/cart-vue.html [hunk 1]
```diff
@@ -45,169 +46,426 @@
...
+            :max-cart-items="100"
+            :grants-by-tenant="grantsByTenant"
+            :grants-under-minimal-contribution="grantsUnderMinimalContribution"
+          >
+            <div>
+              <!-- Checkout with Polygon button -->
+              <button
+                @click="checkoutWithPolygon()"
+                class="btn p-0 checkout-logo"
+                :disabled="grantsUnderMinimalContribution?.length > 0"
```
Comment (nutrina) [inline app/grants/templates/grants/cart-vue.html:267]: @chibie @thelostone-mc @gdixon @ksolo 

I have altered the condition in this disabled attribute (I have moved some verifications to the click handler).
I would also move this verification to the click handler and display an error message if verification fails.

Is there any good reason why not to do this? I mean yes, we'll have to be careful and test the entire flow again, but is there anything particular to consider?
As I have understood from @PixelantDesign yesterday, there where some considerations in the past.

Comment (gdixon) [reply app/grants/templates/grants/cart-vue.html:267]: I think this should be fine, as you say, let's get it tested thoroughly over on staging before we push it out to prod 🙌

**Review by ksolo — APPROVED** [review]

**Review by Jer-Sch — APPROVED** [review]

**Review by chibie — APPROVED** [review]
nice! this looks good to me!

File: app/assets/v2/js/cart.js [hunk 2]
```diff
@@ -28,9 +28,17 @@ const bulkCheckoutAddress = '0x7d655c57f71464B6f83811C55D84009Cd9f5221C';
...
 // Grant data
-let grantHeaders = [ 'Grant', 'Amount', 'Total CLR Match Amount' ]; // cart column headers
 let grantData = []; // data for grants in cart, initialized in mounted hook
 
+Vue.component('eth-checkout-button', {
+  delimiters: [ '[[', ']]' ],
+  template: '#eth-checkout-template',
+  props: [ 'maxCartItems', 'network', 'isZkSyncDown', 'donationInputs', 'onPolygonUpdate', 'currentTokens', 'grantsByTenant',
+    'grantsUnderMinimalContribution', 'isCheckoutOngoing', 'standardCheckout', 'isPolygonDown', 'onZkSyncUpdate' ],
+  }
```
Comment (gdixon) [inline app/assets/v2/js/cart.js:38]: Theres an extraneous closing brace here:
```suggestion
```

**Review by gdixon — CHANGES_REQUESTED** [review]
Apart from 1 tiny bug, this looks and works great! Thanks @nutrina! 🚀

File: app/assets/v2/js/cart.js (hunk 2 above)
Comment (gdixon) [reply app/assets/v2/js/cart.js]: I've committed this - it looks like we introduced it here: c6448e3a71a1c9db2daf54a777e4e3130a70e861

**Review by gdixon — APPROVED** [review]

**Pull request merged**